BitmapUtil/BitmapUtil.cs

```diff
@@ -20,9 +21,18 @@ public class BitmapUtil
             };
 
             var pointer = Marshal.UnsafeAddrOfPinnedArrayElement(image.Data, 0);
-            var bitmap = new Bitmap(image.Width, image.Height, image.Stride, format, pointer);
+            using (var bitmap = new Bitmap(image.Width, image.Height, image.Stride, format, pointer))
+            {
+                var clone = new Bitmap(image.Width, image.Height, format);
+
+                using (var g = Graphics.FromImage(clone))
```

Comment:
Yup, looks like this fixed it, and I honestly don't really know a better way to solve it at this time.
The memory leak issue should be of no concern really, as the map image always get loaded once anyway, and for icons these get cached as needed.